### PR TITLE
Documentation Added

### PR DESCRIPTION
**Pull request by hershd23**
## Description
Documentation of SyferText, currently linked to my localfork(branch tests) can be found here

https://syfertext.readthedocs.io/en/tests/index.html

## Affected Dependencies
The list of dependencies can be found in docs/requirements.txt

## Checklist
- [x] I have followed the [Contribution Guidelines](https://github.com/OpenMined/.github/blob/master/CONTRIBUTING.md) and [Code of Conduct](https://github.com/OpenMined/.github/blob/master/CODE_OF_CONDUCT.md)
- [ ] I have commented my code following the [OpenMined Styleguide](https://github.com/OpenMined/.github/blob/master/STYLEGUIDE.md)
- [ ] I have labeled this PR with the relevant [Type labels](https://github.com/OpenMined/.github/labels?q=Type%3A)
- [ ] My changes are covered by tests

### CODE REVIEW

File: docs/source/conf.py [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+import os
+import sys
+
+sys.path.insert(0, os.path.abspath("../.."))
+sys.setrecursionlimit(1500)
+
+
+# -- Project information -----------------------------------------------------
+
+project = "Syfertext"
```
Comment (Nilanshrajput) [inline docs/source/conf.py:22]:
```suggestion
project = "SyferText"
```

File: syfertext/tokenizer.py [hunk 2]
```diff
@@ -162,9 +162,21 @@ def load_rules(
 
         """
 
-        self.prefixes = prefixes
-        self.suffixes = suffixes
-        self.infixes = infixes
+        # To solve the defult value error in the docs
+        if prefixes:
```
Comment (Nilanshrajput) [inline syfertext/tokenizer.py:166]:
```suggestion
        if prefixes is not None:
```

File: syfertext/tokenizer.py [hunk 3]
```diff
@@ -162,9 +162,21 @@ def load_rules(
...
+            self.prefixes = prefixes
+        else:
+            self.prefixes = TOKENIZER_PREFIXES
+
+        if suffixes:
```
Comment (Nilanshrajput) [inline syfertext/tokenizer.py:171]:
```suggestion
        if suffixes is not None:
```

File: syfertext/tokenizer.py [hunk 4]
```diff
@@ -162,9 +162,21 @@ def load_rules(
...
+            self.suffixes = suffixes
+        else:
+            self.suffixes = TOKENIZER_SUFFIXES
+
+        if infixes:
```
Comment (Nilanshrajput) [inline syfertext/tokenizer.py:176]:
```suggestion
        if infixes is not None:
```

File: syfertext/tokenizer.py [hunk 5]
```diff
@@ -173,7 +185,7 @@ def load_rules(
         if exceptions:
```
Comment (Nilanshrajput) [inline syfertext/tokenizer.py:185]:
```suggestion
        if exceptions is not None:
```

**Comment by Nilanshrajput** [comment]
Nice work, one issue is can you change the order of files and methods as they are defined not alphabetically  

`autodoc_member_order = 'bysource'` https://stackoverflow.com/questions/37209921/python-how-not-to-sort-sphinx-output-in-alphabetical-order

**Comment by hershd23** [comment]
Member order is now done by source and I have also included private and special functions. Take a look once it's built maybe we would not need all those functions to be listed. Let me know if anything has to be excluded from the docs

File: syfertext/tokenizer.py [hunk 6]
```diff
@@ -162,18 +164,30 @@ def load_rules(
 
         """
 
-        self.prefixes = prefixes
-        self.suffixes = suffixes
-        self.infixes = infixes
+        # To solve the defult value error in the docs
```
Comment (AlanAboudib) [inline syfertext/tokenizer.py:167]:
```suggestion
        # If affixes are set to None, they should take the default 
        # values
```

**Review by AlanAboudib — CHANGES_REQUESTED** [review]
Great work @hershd23. Two things to take into account:

- The SyferText version number should appear in the docs
- An issue should be created to proof read the docs and account for some formatting errors and typos. We can go for this as an initial version and improve it iteratively

**Comment by Nilanshrajput** [comment]
@hershd23 one thing, update the readme also, link this documentation website in readme.

**Review by AlanAboudib — APPROVED** [review]
Great work @hershd23

**Pull request merged**